pkg/util/compress/compress.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compress
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha256"
+	"github.com/spf13/afero"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	defaultFS = afero.NewOsFs()
+)
+
+// PackDir will compress a dir to a tar.gz file
+func PackDir(dir string, writer io.Writer) ([]byte, error) {
+	hash := sha256.New()
+	gzWriter := gzip.NewWriter(io.MultiWriter(hash, writer))
+	targzWriter := tar.NewWriter(gzWriter)
+	err := afero.Walk(defaultFS, dir, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.Mode().IsRegular() || info.IsDir() {
+			return nil
+		}
+		header, err := tar.FileInfoHeader(info, info.Name())
+		if err != nil {
+			return err
+		}
+		header.Name = strings.TrimPrefix(strings.ReplaceAll(filePath, dir, ""), string(filepath.Separator))
+		err = targzWriter.WriteHeader(header)
+		if err != nil {
+			return err
+		}
+		f, err := defaultFS.Open(filePath)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		_, err = io.Copy(targzWriter, f)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+	targzWriter.Close()
+	gzWriter.Close()
```

Comment:
Does it also need to close targzWriter and gzWriter if err != nil (line 61)?

Reply:
I will do a double check for this. This part of code is copied, not sure why it's not added in the first place.